poker/src/components/KickModal/KickModal.tsx

```diff
@@ -1,47 +1,104 @@
 import React, { FC } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
+import { INIT_MEMBER } from '../../constants';
 // import GeneralButton from '../GeneralButton/GeneralButton';
-import { isModalCallFromServerSlice, setIsModalOpen } from '../../redux/slices/kickSlice';
+import {
+  isModalOpenBySocketEventSlice,
+  kickIdSlice,
+  setIsModalOpen,
+  setIsModalOpenBySocketEvent,
+  setKickId,
+  setWhoKick,
+  setWhoWillBeKicked,
+  whoKickSlice,
+  whoWillBeKickedSlice,
+} from '../../redux/slices/kickSlice';
 import { SocketContext } from '../../socketContext';
-import { KICKED_MESSAGES, ROLES } from '../../types/common';
-import { mockCurrentUser, mockWhoWillBeKicked, mockRoom } from '../../__mocks__/mockKick';
+import { KICKED_MESSAGES, Member, ROLES } from '../../types/common';
+import { mockRoom } from '../../__mocks__/mockKick';
 import TESTsocket from '../TESTsocket/TESTsocket';
 
 import styles from './KickModal.module.scss';
 
 const KickModal: FC = () => {
-  const currentRoom = mockRoom;
-  const currentUser = mockCurrentUser;
-  const whoWillBeKicked = mockWhoWillBeKicked;
+  const socket = React.useContext(SocketContext);
+
   const dispatch = useDispatch();
 
-  const socket = React.useContext(SocketContext);
+  const isModalOpenBySocketEvent = useSelector(isModalOpenBySocketEventSlice);
+
+  const kickId = useSelector(kickIdSlice);
 
-  const isModalCallFromServer = useSelector(isModalCallFromServerSlice);
+  React.useEffect(() => {
+    const kickUserSuccess = (response: {
+      room: string;
+      whoKick: Member;
+      whoWillBeKicked: Member;
+      kickId: string;
+    }): void => {
+      dispatch(setWhoKick(response.whoKick));
+      dispatch(setWhoWillBeKicked(response.whoWillBeKicked));
+      dispatch(setKickId(response.kickId));
+      dispatch(setIsModalOpenBySocketEvent(true));
+      dispatch(setIsModalOpen(true));
+    };
+
+    socket.on('do-you-want-kick-user', kickUserSuccess);
+    console.log('kick-user-modal isModalOpenBySocketEvent', isModalOpenBySocketEvent);
+
+    return (): void => {
+      socket.off('do-you-want-kick-user', kickUserSuccess);
+    };
+  });
+
+  const currentRoom = mockRoom;
+  const whoKick = useSelector(whoKickSlice);
+  // const whoWillBeKicked = useSelector(whoWillBeKickedSlice);
+  const whoWillBeKicked = {
```

Comment:
Add type to whoWillBeKicked

Reply:
This is a new hit of "Baha Men": 
- Who will be kicked? Who? Who?

Reply:
this variable will be remove after testing